frontend/src/components/MapView/LeftPanel/layersPanel/AnalysisLayerSwitchItem/index.tsx

```diff
@@ -40,11 +43,26 @@ const AnalysisLayerSwitchItem = memo(
     const map = useSelector(mapSelector);
     const [selected, setSelected] = useState<boolean>(true);
     const [isOpacitySelected, setIsOpacitySelected] = useState<boolean>(false);
-    const [opacity, setOpacityValue] = useState<number>(initialOpacity || 0);
+    const opacity = useSelector(opacitySelector('analysis'));
+
+    const setOpacity = useCallback(
+      (value: number) =>
+        dispatch(
+          setStateOpacity({
+            map,
+            value,
+            layerId: 'analysis', // We have to pass undefined here so that the function know that we are in an analysis custom layer
+            layerType: undefined, // We have also to pass undefined here to show that we have analysis custom layer,
```

Comment:
might be worth creating an `analysis` type instead?

Reply:
I am not sure if we need to create an analysis layer type, it maybe worth investigating. For this case i extended `SetOpacityParams` to accept `'analysis'` as a type, so we don't have to use this tricky `undefined` there